src/html/HTMLWriter.jl

```diff
@@ -1848,6 +1862,14 @@ function domify(::DCtx, ::Node, rawnode::Documenter.RawNode)
     return rawnode.name === :html ? DOM.Tag(Symbol("#RAW#"))(rawnode.text) : DOM.Node[]
 end
 
+function generate_index!(ctx::HTMLContext, navnode::Documenter.NavNode)
+    return foreach(getpage(ctx, navnode).mdast.children) do node
+        rec = searchrecord(ctx, navnode, node)
+        if !isnothing(rec)
+            push!(ctx.search_index, rec)
+        end
+    end
```

Comment:
`foreach` doesn't return anything, so it looks a bit weird to me.

```suggestion
    foreach(getpage(ctx, navnode).mdast.children) do node
        rec = searchrecord(ctx, navnode, node)
        if !isnothing(rec)
            push!(ctx.search_index, rec)
        end
    end
    return nothing
```

Also, as a side note: I don't mind having `foreach`-es, but I don't really find them particularly idiomatic. I'd just write this as `for node in getpage(ctx, navnode).mdast.children ... end`.